deps/curl-7.88.1/lib/vtls/sf_ocsp.c

```diff
@@ -266,6 +268,53 @@ static const unsigned int MAX_SLEEP_TIME = 16000000;
 static const int MAX_RETRY = 1;
 
 
+/* Encode request data as a part of URL
+ * Encode all characters reserved by URL
+ * @param url_data The source data needs to be encoded_ptr
+ * @param data_size The size of the source data
+ * @param encoded_ptr The allocated buffer filled with encoded data, need to be freed by caller
+ * return the size of encoded data
+ */
+size_t encodeUrlData(const char *url_data, size_t data_size, char** encoded_ptr)
```

Comment:
what will happen URL is already encoded?

Reply:
That won't happen as here we only encode the base64 request data that will be appended as a part of the URL, not entire URL.
encodeUrlData(ocsp_req_base64, ocsp_req_base64_len, &encoded_ocsp_req_base64);